### DISCUSSION TITLE
i spent a lot of time on the example tutorial. not sure if the plpgsql is correct

### DISCUSSION THREAD

**Discussion by devhandler** [supabase/supabase#15286] [Questions]
see the code below. it's about setting a function to delete storage.

the part confuses me is 
```
  select
      into status, content
           result.status::int, result.content::text
      FROM extensions.http((
```

should it really be the case or should it be
```
  select          result.status::int, result.content::text
      into status, content
      FROM extensions.http((
```

or maybe both are ok? i searched online a lot and didn't get any definitive answer. Also, how should i understand "result" here? i searched postgres official doc, but only found result is a reserved word, but didn't find anything how it should be used, e.g., as a reserved word from result sets returned from extension or something else. (as a comparison, there is good doc about the reserved keyword, new, which is really helpful)

```
-- this is from supabase tutorial
https://supabase.com/docs/guides/getting-started/tutorials/with-svelte

create or replace function delete_storage_object(bucket text, object text, out status int, out content text)
returns record
language 'plpgsql'
security definer
as $$
declare
  project_url text := '<YOURPROJECTURL>';
  service_role_key text := '<YOURSERVICEROLEKEY>'; --  full access needed
  url text := project_url||'/storage/v1/object/'||bucket||'/'||object;
begin
  select
      into status, content
           result.status::int, result.content::text
      FROM extensions.http((
    'DELETE',
    url,
    ARRAY[extensions.http_header('authorization','Bearer '||service_role_key)],
    NULL,
    NULL)::extensions.http_request) as result;
end;
$$;

create or replace function delete_avatar(avatar_url text, out status int, out content text)
returns record
language 'plpgsql'
security definer
as $$
begin
  select
      into status, content
           result.status, result.content
      from public.delete_storage_object('avatars', avatar_url) as result;
end;
$$;

```

**Comment by GaryAustin1** [comment]
Are you having issues with it?
This is the way I deleted a file before the tutorials came out:
https://github.com/orgs/supabase/discussions/3124#discussioncomment-1399576
And a more advanced process here:
https://github.com/GaryAustin1/supa-file-helper

**Reply by ChangoMan** [reply]
I am having issues with the example, when I try to update my avatar, I get `http_request.uri is NULL`

I get the same error when trying to edit my `avatar_url` from the supabase Table editor. Is this some permissions thing I am not setting right?
